stdlib/src/collections/inline_array.mojo

```diff
@@ -122,12 +122,11 @@ struct InlineArray[
             unsafe_assume_initialized: The array of `UnsafeMaybeUninitialized` elements.
         """
 
-        self._array = __mlir_op.`kgen.undef`[_type = Self.type]()
-
-        for i in range(Self.size):
-            unsafe_assume_initialized[i].unsafe_ptr().move_pointee_into(
-                self.unsafe_ptr() + i
-            )
+        self = (
+            UnsafePointer.address_of(unsafe_assume_initialized)
+            .bitcast[Self]()
+            .take_pointee()
+        )
 
```

Comment:
I had to change this initializer to be a direct cast and move, rather than moving each element individually.
Seems like it's correct, but maybe something to check into.